modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -31,7 +32,8 @@ InputParameters validParams<TensorMechanicsAction>()
 }
 
 TensorMechanicsAction::TensorMechanicsAction(const InputParameters & params) :
-    Action(params)
+    Action(params),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : "")
```

Comment:
Good point.  I was thinking of a multi-grain crystal plasticity model which would have separate materials, but the stress divergence will remain the same over the domain.  I'll update this code to remove the base_name.